### PR TITLE
Update template stuff

### LINKED ISSUE #182: no LICENSE.rst file
the license is referenced in the README, but i can't find the licence file itself.

### LINKED ISSUE #183: setup.py develop failures
when doing a fresh install of specutils, i ran ``python setup.py develop`` and it generated a slew of errors that ended with:
```
RecursionError: maximum recursion depth exceeded while calling a Python object
```
immediately re-running the same command then worked as expected. on another computer, ``python setup.py develop`` failed with a different complaint about invalid PYTHONPATH.  running ``python setup.py install`` did work and ``develop`` worked after that.

### PR DESCRIPTION
**Pull request by tepickering**
The PR adds the missing license file and will thus close #182. It also updates ``setup.py`` to the latest template version and updates the ``readthedocs`` python version to 3.6. 

closes #183

### CODE REVIEW

**Comment by nmearl** [comment]
Looks good! Merge when you get the chance.

File: setup.py [hunk 1]
```diff
@@ -66,7 +76,7 @@
 builtins._ASTROPY_PACKAGE_NAME_ = PACKAGENAME
 
 # VERSION should be PEP440 compatible (http://www.python.org/dev/peps/pep-0440)
-VERSION = metadata.get('version', '0.1.dev')
+VERSION = metadata.get('version', '0.0.dev0')
```
Comment (bsipocz) [inline setup.py]: change 0.0 to the next version? Last release was 0.3, so it should be I presume next one will be 0.4?

Comment (nmearl) [reply setup.py]: Good catch!

Comment (nmearl) [reply setup.py]: I do believe that updating in `setup.cfg` and running e.g. `python setup.py develop` will auto-populate the version information in the required places.

Comment (bsipocz) [reply setup.py]: ahh, ok, this is new infrastructure then that I haven't been caught up yet ;)

Comment (nmearl) [reply setup.py]: If you can, go ahead and set the version to `0.3.1` and regenerate. If not, I can do a new PR with the version changes.

File: setup.cfg [hunk 2]
```diff
@@ -47,6 +47,6 @@ edit_on_github = False
 github_project = astropy/specutils
 install_requires = astropy, six, gwcs, scipy
 # version should be PEP440 compatible (http://www.python.org/dev/peps/pep-0440)
-version = 0.1.dev
+version = 0.3.1
```
Comment (bsipocz) [inline setup.cfg]: I believe the dev suffix is still needed though, isn't it?

**Comment by tepickering** [comment]
as far as i have been able to tell with 4 different fresh environments, the ``setup.py`` issues i saw appear to have been fixed. so this PR can close #183.

File: setup.cfg (hunk 2 above)
Comment (tepickering) [reply setup.cfg]: technically it's not, i don't think (``setup.py`` runs fine without it), but might be a process question about how versions get numbered.

**Comment by nmearl** [comment]
Awesome. Merge when you're ready.

**Comment by tepickering** [comment]
i have insufficient power to do so :)

**Pull request merged**